### PR TITLE
[jit] Call mono_register_jit_icall_full () in more places so most jit icalls can be called directly during AOT.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/metadata/marshal.c [hunk 1]
```diff
@@ -121,6 +121,20 @@ get_method_image (MonoMethod *method)
 	return m_class_get_image (method->klass);
 }
 
+#ifdef __cplusplus
+template <typename T>
+static void
+register_icall (T func, const char *name, const char *sigstr, gboolean no_wrapper)
```
[the remark this answers is not in the document]
Comment (vargaz) [reply mono/metadata/marshal.c]: Fixed.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/mini/mini-runtime.c [hunk 2]
```diff
@@ -759,11 +759,21 @@ register_icall_with_wrapper (gpointer func, const char *name, const char *sigstr
...
 }
 
+/*
+ * Register an icall where FUNC is dynamically generated or otherwise not
+ * possible to link to it using NAME during AOT.
+ */
+#ifdef __cplusplus
+template <typename T>
+static void
+register_icall_with_wrapper (T func, const char *name, const char *sigstr)
```
[the remark this answers is not in the document]
Comment (vargaz) [reply mono/mini/mini-runtime.c]: Fixed.

[the remark this answers is not in the document]
Comment (lambdageek) [reply mono/mini/mini-runtime.c]: Still needs one more 😁 
missing the "save" argument.

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**